File: ruby/raindrops/raindrops.rb
# frozen_string_literal: true

# Implementation of the Raindrops exercise in Ruby track on Exercism.
module Raindrops
  # @param number [Integer]
  # @return [String]
  def self.convert(number)
    sounds = ''
    sounds += 'Pling' if (number % 3).zero?
    sounds += 'Plang' if (number % 5).zero?
    sounds += 'Plong' if (number % 7).zero?
    return sounds unless sounds.empty?

    number.to_s
  end
end

File: r/raindrops/raindrops.R
raindrops <- function(number) {
  if (number %% 3 == 0) {
      if ((number/3) %% 5 == 0) {
        if ((number/15) %% 7 == 0) {
          paste("PlingPlangPlong")
        } else {
          paste("PlingPlang")
        }
      } else {
        if ((number/3) %% 7 == 0) {
          paste("PlingPlong")
        } else {
          paste("Pling")
        }
      }
  } else {
    if (number %% 5 == 0) {
      if ((number/5) %% 7 == 0) {
        paste("PlangPlong")
      } else {
        paste("Plang")
      }
    } else {
      if (number %% 7 == 0) {
        paste("Plong")
      } else {
        paste(number)
      }
    }
  }
}
